### PR TITLE
Status, Details, and List rayclusters CLI Functions

### LINKED ISSUE #240: CLI List and view Ray Clusters
Add view status, view details and list rayclusters to CLI

### PR DESCRIPTION
**Pull request by carsonmh**
# Issue link
Closes #240 

# What changes have been made
- Added `list rayclusters` `details raycluster` and `status raycluster` functions in the CLI
- Added `list_clusters_all_namespaces` to list all RayClusters in all the namespaces in a Kubernetes Cluster

# Verification steps
After building SDK, see list of commands using `codeflare`

- `list rayclusters` for listing, optionally specifying namespace or `--all`
- `status raycluster` specifying name as an argument and namespace as an option
- `details raycluster` specifying name as an argument and namespace as an option

There are also unit tests for these functions to be run
## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
Looks great, just a couple of small adjustments:
 - Rather than have the default namespace be "default" for list/status/details, I think we should simply **require** the user to specify the namespace. We are slowly moving away from default="default" (still some things to change in the SDK), as it's not actually what most people use.
  - You may want to clean up output when a cluster of a certain name is not found:
  ```
  Traceback (most recent call last):
  File "/opt/homebrew/bin/codeflare", line 8, in <module>
    sys.exit(cli())
  File "/opt/homebrew/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/meyceoz/Documents/codeflare-sdk/src/codeflare_sdk/cli/commands/status.py", line 19, in raycluster
    cluster = get_cluster(name, namespace)
  File "/Users/meyceoz/Documents/codeflare-sdk/src/codeflare_sdk/cluster/cluster.py", line 433, in get_cluster
    raise FileNotFoundError(
FileNotFoundError: Cluster quicktest is not found in default namespace
```
   This is the correct error, though since this is a CLI, we may want to catch this `FileNotFoundError` and just tell the user the message part "Cluster {name} not found in {ns} namespace"
   - With the current implementation, it seems as though we are rewriting the appwrapper to storage every time we call `status` or `details`, due to recreation of the `Cluster` object. I don't think this is necessarily a blocking issue, but something we may want to think about, or maybe open an issue for, for tracking purposes.

**Review by Maxusmusti — APPROVED** [review]
LGTM

**Comment by MichaelClifford** [comment]
@Maxusmusti @carsonmh  what about using `cluster.get_current_namespace()` instead of requiring a namespace? This should just default to the current context's namespace.

File: src/codeflare_sdk/cli/commands/list.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+def cli():
+    """List a specified resource"""
+    pass
+
+
+@cli.command()
+@click.option("--namespace", type=str)
+@click.option("--all", is_flag=True)
+@click.pass_context
+def rayclusters(ctx, namespace, all):
```
[the remark this answers is not in the document]
Comment (carsonmh) [reply src/codeflare_sdk/cli/commands/list.py:22]: I'll create an alias so the user can do either

File: src/codeflare_sdk/cli/commands/delete.py [hunk 2]
```diff
@@ -13,11 +13,15 @@ def cli():
...
 @click.argument("name", type=str)
-@click.option("--namespace", type=str, default="default")
+@click.option("--namespace", type=str, required=True)
 def raycluster(name, namespace):
     """
     Delete a specified RayCluster from the Kubernetes cluster
     """
-    cluster = get_cluster(name, namespace)
+    try:
+        cluster = get_cluster(name, namespace)
```
[the remark this answers is not in the document]
Comment (carsonmh) [reply src/codeflare_sdk/cli/commands/delete.py:22]: #278

**Pull request merged**